hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/HoodieSparkUtils.scala

```diff
@@ -162,11 +163,11 @@ object HoodieSparkUtils extends SparkAdapterSupport {
       if (rows.isEmpty) {
         Iterator.empty
       } else {
+        val readerAvroSchema = new Schema.Parser().parse(readerAvroSchemaStr)
         val transform: GenericRecord => GenericRecord =
           if (sameSchema) identity
           else {
-            val readerAvroSchema = new Schema.Parser().parse(readerAvroSchemaStr)
-            rewriteRecord(_, readerAvroSchema)
```

Comment:
@xiarixiaoyao since we're changing this, shall we also revisit all the other places that use `rewriteRecord` and consider rebasing them onto the new methods?

Reply:
BTW, one miss for the new API is that previously `rewriteRecord` was validating that the record adheres to the new schema while new method doesn't do that (this obscures the issues when conversion is not following Avro evolution rules)